src/app/api/webhook/webhook.service.ts

```diff
@@ -120,53 +140,82 @@ class WebhookService extends BaseService {
       previousAttributes: { companyId: prevCompanyId },
     } = data
     // If company hasn't been changed - don't bother with any of this
-    if (prevCompanyId === newCompanyId) return
+    if (prevCompanyId === newCompanyId || !prevCompanyId) return
 
     // Only delete prev notifications if client had a valid company before
-    const copilot = new CopilotAPI(this.user.token)
-    const prevCompany = prevCompanyId ? await copilot.getCompany(prevCompanyId) : null
+    let prevCompany, newCompany
+    try {
+      prevCompany = await this.copilot.getCompany(prevCompanyId)
+    } catch (e: unknown) {
+      prevCompany = null
+    }
+    try {
+      newCompany = await this.copilot.getCompany(newCompanyId)
+    } catch (e: unknown) {
+      newCompany = null
+    }
+
+    if (!prevCompany?.name && newCompany?.name) {
+      await this.handleCompanyAssignment(clientId)
+    } else if (!newCompany?.name && prevCompany?.name) {
+      await this.handleCompanyUnassignment(clientId, prevCompanyId)
+    } else if (prevCompany?.name && newCompany?.name) {
```

Comment:
what if prev company name and new company name are same?

Reply:
It doesn't matter because this doesn't check equality, only that both company names aren't falsey.

At the top of this method I check equality using company id instead.